### ISSUE TITLE
Navigation item filter `*` not working

### ISSUE THREAD

**Issue by mwaldmueller** [Icinga/icingaweb2#4772]
## Describe the bug
Navigation item filter for custom variable doesn't work as expected.

## To Reproduce
Host object definitions:
```
object Host "dummyhost1" {
    [...]
    vars.geolocation = "49.453932,11.063591"
}

object Host "dummyhost1" {
    [...]
}
```

Navigation item definition for Host Action:
![hostaction](https://user-images.githubusercontent.com/3973350/166688146-abfc0b3a-5ca2-4bfd-a645-db249440282d.png)

## Expected behavior
Host Action link should only appear at Hosts with custom variable, but appears at all Hosts.

## Your Environment

* Icinga Web 2 version and modules (System - About): 2.10.1
* Web browser used: Chrome
* Icinga 2 version used (`icinga2 --version`): 2.13.3
* PHP version used (`php --version`): 7.3.23
* Server operating system and version: CentOS 7.9

**Comment by sukhwinder33445** [comment]
Hi @mwaldmueller,

I could not reproduce this. Everything is working as expected. Host Action link are only appearing at Hosts with custom variable.

I have added two host objects `Host1` and `Host2` with the custom variable `isOn`. 
`Host1` set to `isOn = "Yes"` and `Host2` set to `isOn = "No"`.

With the `_host_isOn=*` filter, the host action links are added only to hosts that have this customvar set.

# Navigation item definition
<img width="661" alt="Screenshot 2022-05-06 at 16 35 00" src="https://user-images.githubusercontent.com/54990055/167154635-37542d01-5b8e-4c1d-be9f-22e89bddcff6.png">


If it is still a bug, can you please add more information on how to reproduce the issue?

**Comment by lippserd** [comment]
> `Host1` set to `isOn = "Yes"` and `Host2` set to `isOn = "No"`.

You set the custom variable on both hosts. Set it only on one host.

**Comment by sukhwinder33445** [comment]
Removed the customvar `isOn` from `Host2`. Now the Host Action link is only visible in the detail view of "Host1".

**Comment by mwaldmueller** [comment]
@sukhwinder33445 I think I already shared the details that are relevant, what else do you need?

**Comment by sukhwinder33445** [comment]
Bug exists before commit `83567ebad`.

**Comment by raviks789** [comment]
Earlier the `FilterExpression::matches()` used to incorrectly match the row even though the value for the column was set null. As the pattern used, would always give a match (even when the value is null). Hence, Action::getRender() method used to return true even when the custom variable for the object was not defined. As a result, the action link used to be created not only in the host/service where the custom variable is defined, but also for other hosts/services where it was not defined because that custom variable would be treated as a column in `FilterExpression::matches()`. But that is resolved through this commit [3dc384fb5](https://github.com/Icinga/icingaweb2/commit/83567ebad2411a1f17819c174492c255e401c96b#diff-944745dec630ccc49611e71311d597202ee84caf0192d9e4a7d98614e32def97).
